### PR TITLE
Make Transit Gateway support Domain Manager accounts

### PR DESCRIPTION
**Pull request by dav3r**
#  #

## 🗣 Description ##

This PR adds the newly-created Domain Manager accounts to the list of those accounts that are allowed to access our Transit Gateway.

## 💭 Motivation and context ##

Without this, the Domain Manager accounts would not be able to be included in the COOL network.

This work is part of https://github.com/cisagov/cool-system/issues/114.

## 🧪 Testing ##

I applied this change in the Shared Services (Staging) environment and verified that the expected Transit Gateway resources were created for the Domain Manager (Staging) account.

Further testing will be conducted with @zenine07 to confirm that Domain Manager is able to successfully route traffic through the Transit Gateway.

## ✅ Checklist ##

* [x] This PR has an informative and human-readable title.
* [x] Changes are limited to a single goal - _eschew scope creep!_
* [x] All relevant type-of-change labels have been added.
* [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
* [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
* [x] All new and existing tests pass.

## ✅  Post-Merge Checklist ##
* [x] Apply Terraform in Production workspace

### CODE REVIEW

File: transitgateway.tf [hunk 1]
```diff
@@ -44,13 +44,13 @@ resource "aws_ram_resource_association" "tgw" {
...
-# Share the resource with the other accounts that are allowed to
-# access it (currently the env*, PCA, and User Services accounts).
+# Share the resource with the other accounts that are allowed to access
+# it (currently the Domain Manger, env*, PCA, and User Services accounts).
 #
 resource "aws_ram_principal_association" "tgw" {
   provider = aws.sharedservicesprovisionaccount
 
-  for_each = merge(local.env_accounts_same_type, local.pca_account_same_type, local.userservices_account_same_type)
+  for_each = merge(local.domainmanager_account_same_type, local.env_accounts_same_type, local.pca_account_same_type, local.userservices_account_same_type)
```
Comment (jsf9k) [inline transitgateway.tf:53]: I was going to suggest that we create a `local` variable for this `merge()`, but then I realized that `local.userservices_account_same_type` isn't listed in the two `merge()`s below this.  Can you remind me why not?

**Review by jsf9k — CHANGES_REQUESTED** [review]
Not really requesting changes, but I have a question.

File: transitgateway.tf (hunk 1 above)
Comment (dav3r) [reply transitgateway.tf:53]: The answer you seek lies in this code comment: https://github.com/cisagov/cool-sharedservices-networking/blob/b76bd3a963e1a140dbbada41b547d661c418d88d/transitgateway.tf#L64-L66

It came about from [this PR comment thread](https://github.com/cisagov/cool-sharedservices-networking/pull/37#pullrequestreview-580719757).

Comment (jsf9k) [reply transitgateway.tf:53]: Of course I know him.  He's me!

Thanks!

**Review by jsf9k — APPROVED** [review]

**Review by hillaryj — APPROVED** [review]
🌏

**Pull request merged**